crawler/src/main/java/com/movie/recommender/crawler/dataimport/MovieDataInitializer.java

```diff
@@ -10,19 +12,14 @@
 @Slf4j
 public class MovieDataInitializer {
     private final MovieDataLoader movieDataLoader;
-    private final MongoDBService mongoDBService;
 
-    public MovieDataInitializer(MovieDataLoader movieDataLoader, MongoDBService mongoDBService) {
+    public MovieDataInitializer(MovieDataLoader movieDataLoader, MovieRepository movieRepository) {
         this.movieDataLoader = movieDataLoader;
-        this.mongoDBService = mongoDBService;
     }
 
-    public void initializeData(String filePath) {
-        // Load data using MovieDataLoader
+    public List<MovieDetails>  initializeData(String filePath) {
         List<MovieDetails> movieDetailsList = movieDataLoader.parseMovieData(filePath);
         log.info("Loaded {} movie records from {}", movieDetailsList.size(), filePath);
-
-        // Insert data into MongoDB
-        mongoDBService.insertMovies(movieDetailsList);
+        return movieDetailsList;
```

Comment:
call `mongoDBService.insertMovies(movieDetailsList)` here before return, not in `Main`